pkg/daemon/addr/utils.go

```diff
@@ -0,0 +1,87 @@
+package addr
+
+import (
+	"fmt"
+	"net"
+
+	"github.com/vishvananda/netlink"
+	"golang.org/x/sys/unix"
+)
+
+func checkIfEnhancedAddr(link netlink.Link, addr netlink.Addr, family int) (bool, error) {
+	routeList, err := netlink.RouteListFiltered(family, &netlink.Route{
+		Table:     unix.RT_TABLE_LOCAL,
+		LinkIndex: link.Attrs().Index,
+		Src:       addr.IP,
+	}, netlink.RT_FILTER_TABLE|netlink.RT_FILTER_OIF|netlink.RT_FILTER_SRC)
+
+	if err != nil {
+		return false, fmt.Errorf("list local routes for interface %v and src %v failed: %v",
+			link.Attrs().Name, addr.IP.String(), err)
+	}
+
+	if len(routeList) == 0 && addr.Flags&unix.IFA_F_SECONDARY == 0 {
```

Comment:
If enhanced address is what we need for the source IP of ARP request, I think we miss something here.

Reply:
If we are using the bridge mode, then the original IP of same subnet can be used for ARP request, but it won't hit this judgment.

Reply:
fixed~